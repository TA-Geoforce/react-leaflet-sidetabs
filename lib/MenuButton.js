"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var MenuButton = function MenuButton(props) {
  var icon = props.icon === 'string' ? /*#__PURE__*/_react.default.createElement("i", {
    className: props.icon
  }) : props.icon;
  var active = props.id === props.selected && !props.collapsed ? ' active' : '';
  var glowing = props.id === props.glowing ? ' glowing' : '';
  var disabled = props.disabled ? ' disabled' : '';
  return /*#__PURE__*/_react.default.createElement("li", {
    className: active + disabled + glowing,
    key: props.id
  }, /*#__PURE__*/_react.default.createElement("a", {
    href: "#" + props.id,
    role: "tab",
    onClick: function onClick(e) {
      return props.disabled || (props.collapsed ? props.onOpen(props.id) : props.selected === props.id ? props.onClose() : props.onOpen(props.id));
    }
  }, ' ', icon));
};

MenuButton.propTypes = {
  id: _propTypes.PropTypes.string.isRequired,
  icon: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.string, _propTypes.PropTypes.element]).isRequired,
  disabled: _propTypes.PropTypes.bool,
  selected: _propTypes.PropTypes.string,
  glowing: _propTypes.PropTypes.string,
  onOpen: _propTypes.PropTypes.func,
  onClose: _propTypes.PropTypes.func,
  collapsed: _propTypes.PropTypes.bool
};
var _default = MenuButton;
exports.default = _default;