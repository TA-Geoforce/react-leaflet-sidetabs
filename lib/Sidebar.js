"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _Tab = _interopRequireDefault(require("./Tab"));

var _MenuButton = _interopRequireDefault(require("./MenuButton"));

require("../sidebar.css");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var TabType = _propTypes.PropTypes.shape({
  type: _propTypes.PropTypes.oneOf([_Tab.default])
});

var Sidebar = /*#__PURE__*/function (_Component) {
  (0, _inheritsLoose2.default)(Sidebar, _Component);

  function Sidebar() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = Sidebar.prototype;

  _proto.onClose = function onClose(e) {
    e.preventDefault();
    e.stopPropagation();
    this.props.onClose && this.props.onClose(e);
  };

  _proto.onOpen = function onOpen(e, tabid) {
    e.preventDefault();
    e.stopPropagation();
    this.props.onOpen && this.props.onOpen(tabid);
  };

  _proto.renderPanes = function renderPanes(children) {
    var _this = this;

    return _react.default.Children.map(children, function (p) {
      return /*#__PURE__*/_react.default.cloneElement(p, {
        onClose: _this.onClose.bind(_this),
        closeIcon: _this.props.closeIcon,
        active: p.props.id === _this.props.selected,
        position: _this.props.position || 'left'
      });
    });
  };

  _proto.render = function render() {
    var _this2 = this;

    var position = " sidebar-" + (this.props.position || 'left');
    var collapsed = this.props.collapsed ? ' collapsed' : '';

    var tabs = _react.default.Children.toArray(this.props.children);

    var bottomtabs = tabs.filter(function (t) {
      return t.props.anchor === 'bottom';
    });
    var toptabs = tabs.filter(function (t) {
      return t.props.anchor !== 'bottom';
    });
    return /*#__PURE__*/_react.default.createElement("div", {
      id: this.props.id,
      className: "sidebar leaflet-touch" + position + collapsed,
      ref: function ref(el) {
        _this2.rootElement = el;
      }
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "sidebar-tabs"
    }, /*#__PURE__*/_react.default.createElement("ul", {
      role: "tablist"
    }, toptabs.map(function (t) {
      return /*#__PURE__*/_react.default.createElement(_MenuButton.default, {
        key: t.props.id,
        id: t.props.id,
        icon: t.props.icon,
        disabled: t.props.disabled,
        selected: _this2.props.selected,
        glowing: _this2.props.glowing,
        collapsed: _this2.props.collapsed,
        onClose: _this2.props.onClose,
        onOpen: _this2.props.onOpen
      });
    })), /*#__PURE__*/_react.default.createElement("ul", {
      role: "tablist"
    }, bottomtabs.map(function (t) {
      return /*#__PURE__*/_react.default.createElement(_MenuButton.default, {
        key: t.props.id,
        id: t.props.id,
        icon: t.props.icon,
        disabled: t.props.disabled,
        selected: _this2.props.selected,
        glowing: _this2.props.glowing,
        collapsed: _this2.props.collapsed,
        onClose: _this2.props.onClose,
        onOpen: _this2.props.onOpen
      });
    }))), /*#__PURE__*/_react.default.createElement("div", {
      className: "sidebar-content"
    }, this.renderPanes(this.props.children)));
  };

  return Sidebar;
}(_react.Component);

Sidebar.propTypes = {
  id: _propTypes.PropTypes.string.isRequired,
  collapsed: _propTypes.PropTypes.bool,
  position: _propTypes.PropTypes.oneOf(['left', 'right']),
  selected: _propTypes.PropTypes.string,
  glowing: _propTypes.PropTypes.string,
  closeIcon: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.string, _propTypes.PropTypes.element]),
  onClose: _propTypes.PropTypes.func,
  onOpen: _propTypes.PropTypes.func,
  children: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.arrayOf(TabType), TabType])
};
var _default = Sidebar;
exports.default = _default;