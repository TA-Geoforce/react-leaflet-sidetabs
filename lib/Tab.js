"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = require("prop-types");

var Tab = function Tab(props) {
  var active = props.active ? ' active' : '';
  var closeIcon = closeIconSelector(props);
  return /*#__PURE__*/_react.default.createElement("div", {
    id: props.id,
    className: "sidebar-pane" + active
  }, /*#__PURE__*/_react.default.createElement("h1", {
    className: "sidebar-header"
  }, props.header, /*#__PURE__*/_react.default.createElement("div", {
    className: "sidebar-close",
    role: "btn",
    onClick: props.onClose
  }, closeIcon)), props.children);
};

var closeIconSelector = function closeIconSelector(props) {
  switch (typeof props.closeIcon) {
    case 'string':
      return /*#__PURE__*/_react.default.createElement("i", {
        className: props.closeIcon
      });

    case 'object':
      return props.closeIcon;

    default:
      return props.position === 'right' ? /*#__PURE__*/_react.default.createElement("i", {
        className: "fa fa-caret-right"
      }) : /*#__PURE__*/_react.default.createElement("i", {
        className: "fa fa-caret-left"
      });
  }
};

closeIconSelector.propTypes = {
  closeIcon: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.string, _propTypes.PropTypes.element]),
  position: _propTypes.PropTypes.oneOf(['left', 'right'])
};
Tab.propTypes = {
  id: _propTypes.PropTypes.string.isRequired,
  header: _propTypes.PropTypes.string.isRequired,
  children: _propTypes.PropTypes.oneOfType([_propTypes.PropTypes.func, _propTypes.PropTypes.element, _propTypes.PropTypes.object]).isRequired,
  onClose: _propTypes.PropTypes.func,
  active: _propTypes.PropTypes.bool
};
var _default = Tab;
exports.default = _default;