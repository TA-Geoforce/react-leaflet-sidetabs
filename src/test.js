import React from 'react'
import renderer from 'react-test-renderer'
import MenuButton from './MenuButton'
import Tab from './Tab'
import Sidebar from './Sidebar'
describe('MenuButton', () => {
  it('is truthy', () => {
    expect(MenuButton).toBeTruthy()
  })

  it('has not active class when selected not equals id', () => {
    const tree = renderer.create(
      <MenuButton
        key='test'
        id='test'
        icon='fa fa-home'
        disabled={false}
        selected='nottest'
        collapsed={true}
        onClose={() => {}}
        onOpen={() => {}} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has active class when selected equals id', () => {
    const tree = renderer.create(
      <MenuButton
        key='test'
        id='test'
        icon='fa fa-home'
        disabled={false}
        selected='test'
        collapsed={false}
        onClose={() => {}}
        onOpen={() => {}} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has disabled class disabled equals true', () => {
    const tree = renderer.create(
      <MenuButton
        key='test'
        id='test'
        icon='fa fa-home'
        disabled={true}
        selected='test'
        collapsed={false}
        onClose={() => {}}
        onOpen={() => {}} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Tab', () => {
  it('is truthy', () => {
    expect(Tab).toBeTruthy()
  })

  it('has not active class without active prop', () => {
    const tree = renderer.create(
      <Tab id='home' header='Home' icon='fa fa-home'>
        <p>No place like home!</p></Tab>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has active class with active prop', () => {
    const tree = renderer.create(
      <Tab id='home' header='Home' active={true} icon='fa fa-home'>
        <p>No place like home!</p></Tab>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('makes closeIcon fa fa-caret-right when position is right', () => {
    const tree = renderer.create(
      <Tab id='home' header='Home' active={true} position='right' icon='fa fa-home'>
        <p>No place like home!</p></Tab>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('makes closeIcon fa fa-caret-left when position is left', () => {
    const tree = renderer.create(
      <Tab id='home' header='Home' active={true} position='left' icon='fa fa-home'>
        <p>No place like home!</p></Tab>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Sidebar', () => {
  it('is truthy', () => {
    expect(Sidebar).toBeTruthy()
  })
})
