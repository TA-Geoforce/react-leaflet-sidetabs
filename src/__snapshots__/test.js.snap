// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton has active class when selected equals id 1`] = `
<li
  className=" active"
>
  <a
    href="#test"
    onClick={[Function]}
    role="tab"
  >
     
    fa fa-home
  </a>
</li>
`;

exports[`MenuButton has disabled class disabled equals true 1`] = `
<li
  className=" active disabled"
>
  <a
    href="#test"
    onClick={[Function]}
    role="tab"
  >
     
    fa fa-home
  </a>
</li>
`;

exports[`MenuButton has glowing on 1`] = `
<li
  className=" active disabled"
>
  <a
    href="#test"
    onClick={[Function]}
    role="tab"
  >
     
    fa fa-home
  </a>
</li>
`;

exports[`MenuButton has not active class when selected not equals id 1`] = `
<li
  className=""
>
  <a
    href="#test"
    onClick={[Function]}
    role="tab"
  >
     
    fa fa-home
  </a>
</li>
`;

exports[`Tab has active class with active prop 1`] = `
<div
  className="sidebar-pane active"
  id="home"
>
  <h1
    className="sidebar-header"
  >
    Home
    <div
      className="sidebar-close"
      onClick={undefined}
      role="btn"
    >
      <i
        className="fa fa-caret-left"
      />
    </div>
  </h1>
  <p>
    No place like home!
  </p>
</div>
`;

exports[`Tab has not active class without active prop 1`] = `
<div
  className="sidebar-pane"
  id="home"
>
  <h1
    className="sidebar-header"
  >
    Home
    <div
      className="sidebar-close"
      onClick={undefined}
      role="btn"
    >
      <i
        className="fa fa-caret-left"
      />
    </div>
  </h1>
  <p>
    No place like home!
  </p>
</div>
`;

exports[`Tab makes closeIcon fa fa-caret-left when position is left 1`] = `
<div
  className="sidebar-pane active"
  id="home"
>
  <h1
    className="sidebar-header"
  >
    Home
    <div
      className="sidebar-close"
      onClick={undefined}
      role="btn"
    >
      <i
        className="fa fa-caret-left"
      />
    </div>
  </h1>
  <p>
    No place like home!
  </p>
</div>
`;

exports[`Tab makes closeIcon fa fa-caret-right when position is right 1`] = `
<div
  className="sidebar-pane active"
  id="home"
>
  <h1
    className="sidebar-header"
  >
    Home
    <div
      className="sidebar-close"
      onClick={undefined}
      role="btn"
    >
      <i
        className="fa fa-caret-right"
      />
    </div>
  </h1>
  <p>
    No place like home!
  </p>
</div>
`;
